client/cmd.go

```diff
@@ -251,6 +251,11 @@ func readTxCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, err
 		}
 	}
 
+	if !clientCtx.IsAux || flagSet.Changed(flags.FlagAux) {
+		isAux, _ := flagSet.GetBool(flags.FlagAux)
+		clientCtx = clientCtx.WithAux(isAux)
```

Comment:
It's a bit weird that if you only add `--aux`, then it errors.

@atheeshp What do you think of adding:

```go
		if isAux {
			// If the user didn't explicity set an --output flag, use JSON by
			// default.
			if clientCtx.OutputFormat == "" || !flagSet.Changed(cli.OutputFlag) {
				clientCtx = clientCtx.WithOutputFormat("json")
			}

			// If the user didn't explicity set a --sign-mode flag, use
			// DIRECT_AUX by default.
			if clientCtx.SignModeStr == "" || !flagSet.Changed(flags.FlagSignMode) {
				clientCtx = clientCtx.WithSignModeStr(flags.SignModeDirectAux)
			}
		}
```